soc/riscv/riscv-privilege/telink_b91/power.c

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (c) 2022 Telink Semiconductor
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <ext_driver/ext_pm.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/pm/pm.h>
+#include <zephyr/pm/device.h>
+#include <stimer.h>
+#include <zephyr/zephyr.h>
+
+LOG_MODULE_DECLARE(soc, CONFIG_SOC_LOG_LEVEL);
+
+#define DT_DRV_COMPAT telink_machine_timer
+
+#define MTIME_REG	DT_INST_REG_ADDR(0)
+#define MTIMECMP_REG	(DT_INST_REG_ADDR(0) + 8)
+
+#define mticks_to_systicks(mticks)                                                                 \
+	(((uint64_t)(mticks)*SYSTEM_TIMER_TICK_1S) / CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC)
+#define systicks_to_mticks(sticks)                                                                 \
+	(((uint64_t)(sticks)*CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC) / SYSTEM_TIMER_TICK_1S)
```

Comment:
why isn't this using tick conversion utilities?

Reply:
systicks is not standard risc-v mtimer ticks. it's specific platform ticks which are used duting sleep mode.
so we are converting risc-v mtimer ticks (32K) to platform specific ticks (16M). it's not converting mtimer ticks to mS